### PR TITLE
Output all empty snapshot info fields if in verbose mode

### PR DESCRIPTION
**Pull request by abeyad**
In #24477, a less verbose option was added to retrieve snapshot info via
GET /_snapshot/{repo}/{snapshots}.  The point of adding this less
verbose option was so that if the repository is a cloud based one, and
there are many snapshots for which the snapshot info needed to be
retrieved, then each snapshot would require reading a separate snapshot
metadata file to pull out the necessary information.  This can be costly
(performance and cost) on cloud based repositories, so a less verbose
option was added that only retrieves very basic information about each
snapshot that is all available in the index-N blob - requiring only one
read!

In order to display this less verbose snapshot info appropriately, logic
was added to not display those fields which could not be populated.
However, this broke integrators (e.g. ECE) that required these fields to
be present, even if empty.  This commit is to return these fields in the
response, even if empty, if the verbose option is set.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetSnapshotsAction.java [hunk 1]
```diff
@@ -52,10 +60,19 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
...
+        final GetSnapshotsRequest getSnapshotsRequest = getSnapshotsRequest(repository).snapshots(snapshots);
         getSnapshotsRequest.ignoreUnavailable(request.paramAsBoolean("ignore_unavailable", getSnapshotsRequest.ignoreUnavailable()));
         getSnapshotsRequest.verbose(request.paramAsBoolean("verbose", getSnapshotsRequest.verbose()));
         getSnapshotsRequest.masterNodeTimeout(request.paramAsTime("master_timeout", getSnapshotsRequest.masterNodeTimeout()));
-        return channel -> client.admin().cluster().getSnapshots(getSnapshotsRequest, new RestToXContentListener<>(channel));
+        return channel -> client.admin().cluster().getSnapshots(getSnapshotsRequest,
+            new RestBuilderListener<GetSnapshotsResponse>(channel) {
+                @Override
+                public RestResponse buildResponse(GetSnapshotsResponse response, XContentBuilder builder) throws Exception {
+                    Map<String, String> params = Collections.singletonMap("verbose", Boolean.toString(getSnapshotsRequest.verbose()));
```
Comment (imotov) [inline core/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetSnapshotsAction.java]: I don't think is necessary `request` is passed as`params`. So, "verbose" is already in `params` if it was specified, you just need to be careful resolving defaults.

**Review by imotov — APPROVED** [review]
Left one comment regarding dealing with params. I think it can be simplified. Otherwise, LGTM.

**Comment by abeyad** [comment]
I pushed 3d34d08dcaa1244a3aacc306025bdf3f64c3e045 to address your comment.  Thanks for the review @imotov!

**Comment by rjernst** [comment]
I understand the reason for this remaining the "default" behavior in 5.x, so as not to break compatibility within a major, but we should still be able to remove this long term? As a followup can the shorter form be used when a url param is set, and that then become the default in 6.0?

**Comment by abeyad** [comment]
@rjernst I had a discussion with @clintongormley about this a few weeks ago and thought maybe we should make the default verbose=false for 6.x, but didn't reach a decision at the time.  @clintongormley what do you think?

**Comment by abeyad** [comment]
5.x commit: 950123320e20359290c94c50ac5f7d34d54dfa4d
5.5 commit: 3afbafa13067e9795cea9aa92f3e1ac39fab64d3

**Comment by clintongormley** [comment]
> thought maybe we should make the default verbose=false for 6.x, but didn't reach a decision at the time.

we could do this quite cleanly - if no `verbose` flag is specified, then we can emit a deprecation warning saying they should use `verbose=true` going forwards, then change the default in 6.0

**Pull request merged**